crates/uv-resolver/src/lock/mod.rs

```diff
@@ -633,17 +633,26 @@ impl Lock {
             }
 
             // Add any dev dependencies.
-            for group in dev.iter() {
-                for dep in root.dependency_groups.get(group).into_iter().flatten() {
-                    if dep.complexified_marker.evaluate(marker_env, &[]) {
-                        let dep_dist = self.find_by_id(&dep.package_id);
-                        if seen.insert((&dep.package_id, None)) {
-                            queue.push_back((dep_dist, None));
-                        }
-                        for extra in &dep.extra {
-                            if seen.insert((&dep.package_id, Some(extra))) {
-                                queue.push_back((dep_dist, Some(extra)));
-                            }
+            for dep in root
```

Comment:
I'm struggling to find a straightforward way to make `dev.iter()` work with `IncludeGroups::All`, as it needs to be resolved to specific groups before iteration. Consequently, I changed `dev.iter()` to `dev.contains(group)`, allowing `DevGroupManifest` to decide which groups are included. Hopefully, there is a more elegant solution than this.